### PR TITLE
Add support for skills

### PR DESCRIPTION
**Pull request by kcotugno**

### CODE REVIEW

File: lib/config/specification.yml [hunk 1]
```diff
@@ -746,6 +746,67 @@ roles: # TODO(SZ): find it out
...
+    skill:
+      foreign_key: skill_id
+      collection: skills
+    user:
+      foreign_key: user_id
+      collection: users
+    creator:
+      foreign_key: creator_id
+      collection: users
+
```
Comment (arincon9) [inline lib/config/specification.yml]: ❌Remove space

File: lib/config/specification.yml [hunk 2]
```diff
@@ -746,6 +746,67 @@ roles: # TODO(SZ): find it out
...
+    - max_level
+    - description
+    - role_ids
+
+skill_categories:
+  associations:
+    skills:
+      foreign_key: skill_ids
+      collection: skills
+
```
Comment (arincon9) [inline lib/config/specification.yml]: ❌Remove space

File: lib/config/specification.yml [hunk 3]
```diff
@@ -746,6 +746,67 @@ roles: # TODO(SZ): find it out
...
 
+skills:
+  associations:
+    skill_category:
+      foreign_key: skill_category_id
+      collection: skill_categories
+    roles:
+      foreign_key: role_ids
+      collection: roles
+
```
Comment (arincon9) [inline lib/config/specification.yml]: ❌Remove space

**Comment by arincon9** [comment]
👍  LGTM

**Comment by kcotugno** [comment]
Thanks @arincon9.

**Pull request merged**